src/diffix/DConnection.cs

```diff
@@ -1,13 +1,16 @@
 namespace Diffix
 {
+    using System;
     using System.Threading.Tasks;
 
     public interface DConnection
     {
+        bool IsCancellationRequested { get; }
+
         Task<DResult<TRow>> Exec<TRow>(DQuery<TRow> query);
 
         void Cancel();
 
-        void ThrowIfCancellationRequested();
+        void ThrowIfCancellationRequested() => throw new OperationCanceledException("Query operation was cancelled.");
```

Comment:
This will throw even if no cancellation is requested. It should be:

```suggestion
        void ThrowIfCancellationRequested() 
        {
            if (IsCancellationRequested)
            {
                throw new OperationCanceledException("Query operation was cancelled.");
            }
        }
```

However maybe it's simpler to just expose a `CancellationToken`...